apps/io-wallet-user-func/src/infra/azure/cosmos/wallet-instance.ts

```diff
@@ -83,28 +80,20 @@ export class CosmosDbWalletInstanceRepository
 
   getAllByUserId(userId: WalletInstance["userId"]) {
     return pipe(
-      TE.tryCatch(
-        async () => {
-          const { resources: items } = await this.#container.items
-            .query({
-              parameters: [
-                {
-                  name: "@partitionKey",
-                  value: userId,
-                },
-              ],
-              query: "SELECT * FROM c WHERE c.userId = @partitionKey",
-            })
-            .fetchAll();
-          return items;
-        },
-        (error) =>
-          error instanceof Error && error.name === "TimeoutError"
-            ? new ServiceUnavailableError(
-                `The request to the database has timed out: ${error.message}`,
-              )
-            : new Error(`Error getting wallet instances by user id: ${error}`),
-      ),
+      TE.tryCatch(async () => {
+        const { resources: items } = await this.#container.items
+          .query({
+            parameters: [
+              {
+                name: "@partitionKey",
+                value: userId,
+              },
+            ],
+            query: "SELECT * FROM c WHERE c.userId = @partitionKey",
+          })
+          .fetchAll();
```

Comment:
What about returning an iterator here instead so you can traverse the whole collection without recursive calls?

Reply:
Right, I introduced the `AsyncIterator` even if it's not the best for using with `ReaderTaskEither` and `TaskEither`. Take a look now.